### PR TITLE
[Neo Core Storage] Implicit methods and tests

### PR DESCRIPTION
**Pull request by Jim8y**
# Description

This pr adds implicit methods for StorageItem and StorageKey, and also added Unit Test to test them.

Fixes # (issue)

## Type of change

- [x] Optimization (the change is only an optimization)
- [ ] Style (the change is only a code style for better maintenance or standard purpose)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] TestStorageKey
- [x] TestStorageItem

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Neo/SmartContract/StorageKey.cs [hunk 1]
```diff
@@ -79,5 +80,17 @@ public byte[] ToArray()
...
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static implicit operator StorageKey(byte[] value)
+        {
+            return new StorageKey(value);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static implicit operator StorageKey(ReadOnlyMemory<byte> value)
+        {
+            return value.ToArray();
```
Comment (cschuchardt88) [inline src/Neo/SmartContract/StorageKey.cs:93]:
```suggestion
            return value.Span;
```

`Span` is faster and less memory. I don't think you need to add `Span` to `ctor`, but we should be using `Span<>`, `ReadOnlySpan<>` for non-threaded and `ReadOnlyMemory<>` and `Memory<>` for `async`

**Review by shargon — APPROVED** [review]

File: src/Neo/SmartContract/StorageItem.cs [hunk 2]
```diff
@@ -193,5 +193,15 @@ public void Set(IInteroperable interoperable)
             item.cache ??= new BigInteger(item.value.Span);
             return (BigInteger)item.cache;
         }
+
+        public static implicit operator StorageItem(BigInteger value)
```
Comment (AnnaShaleva) [inline src/Neo/SmartContract/StorageItem.cs:197]: I'm just wondering, do you have a use-case for these conversion operators in the code? Usually when we introduce such helpers, there's a need to use them in the code and we replace some real code with these helpers.

Comment (cschuchardt88) [reply src/Neo/SmartContract/StorageItem.cs:197]: Doesn't matter how it done. As long as the do the same thing. We have direct access to the `class`. So, no need for extension methods.

Comment (Jim8y) [reply src/Neo/SmartContract/StorageItem.cs:197]: > I'm just wondering, do you have a use-case for these conversion operators in the code? Usually when we introduce such helpers, there's a need to use them in the code and we replace some real code with these helpers.

Does it matter to you?

**Review by cschuchardt88 — APPROVED** [review]

**Pull request merged**